packages/app/src/components-styled/time-series-chart/components/series.tsx

```diff
@@ -49,54 +50,74 @@ function SeriesUnmemoized<T extends TimestampedValue>({
-                color={config.color}
-                style={config.style}
-                strokeWidth={config.strokeWidth}
-                getX={getX}
-                getY={getY}
-                onHover={(evt) => onHover && onHover(evt, index)}
-              />
-            );
-          case 'area':
-            return (
-              <AreaTrend
-                key={index}
-                series={series as SeriesSingleValue[]}
-                color={config.color}
-                style={config.style}
-                fillOpacity={config.fillOpacity}
-                strokeWidth={config.strokeWidth}
-                getX={getX}
-                getY={getY}
-                yScale={yScale}
-                onHover={(evt) => onHover && onHover(evt, index)}
-              />
-            );
-
-          case 'range':
-            return (
-              <RangeTrend
-                key={config.metricPropertyLow as string}
-                series={series as SeriesDoubleValue[]}
-                color={config.color}
-                fillOpacity={config.fillOpacity}
-                getX={getX}
-                getY0={getY0}
-                getY1={getY1}
-                bounds={bounds}
-              />
-            );
-        }
-      })}
+          switch (config.type) {
+            case 'line':
+              return (
+                <LineTrend
+                  key={config.metricProperty as string}
+                  series={series as SeriesSingleValue[]}
+                  color={config.color}
+                  style={config.style}
+                  strokeWidth={config.strokeWidth}
+                  getX={getX}
+                  getY={getY}
+                  onHover={(evt) => onHover && onHover(evt, index)}
+                />
+              );
+            case 'area':
+              return (
+                <AreaTrend
+                  key={index}
+                  series={series as SeriesSingleValue[]}
+                  color={config.color}
+                  style={config.style}
+                  fillOpacity={config.fillOpacity}
+                  strokeWidth={config.strokeWidth}
+                  getX={getX}
+                  getY={getY}
+                  yScale={yScale}
+                  onHover={(evt) => onHover && onHover(evt, index)}
+                />
+              );
+            case 'range':
+              return (
+                <RangeTrend
+                  key={config.metricPropertyLow as string}
+                  series={series as SeriesDoubleValue[]}
+                  color={config.color}
+                  fillOpacity={config.fillOpacity}
+                  getX={getX}
+                  getY0={getY0}
+                  getY1={getY1}
+                  bounds={bounds}
+                />
+              );
+            case 'stacked-area':
+              return (
+                <StackedAreaTrend
+                  key={`${config.metricProperty}${index}` as string}
```

Comment:
casting is not needed per se, I guess?

```suggestion
                  key={`${config.metricProperty}${index}`}
```